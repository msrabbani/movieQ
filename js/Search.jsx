import React, {Component} from 'react';
import DataPilem from '../data.json';
import ShowCard from './ShowCard';

class Search extends Component {
  /* constructor(props) {
    super(props);

    this.state = {
      searchTerm: 'this is some sort debug statment'
    };

    this.handleSearchTermChange = this.handleSearchTermChange.bind(this);
  }

  handleSearchTermChange(event) {
    this.setState({ searchTerm: event.target.value });
  }*/

  state = {
    searchTerm: ''
  };
  handleSearchTermChange = event => {
    this.setState({searchTerm: event.target.value});
  };

  render() {
    return (
      <div className="search">
        {/* <pre><code> {JSON.stringify(DataPilem, null,4)} </code></pre>*/}

        {/* {DataPilem.shows.map(show => <h3>{show.title}</h3> )}*/}

        <header>
            <h1>videoQ</h1>
           <h1>{this.state.searchTerm.toUpperCase()}</h1>

          <input
            onChange={this.handleSearchTermChange}
            value={this.state.searchTerm}
            type="text"
            placeholder="Search"
          />
        </header>
        <div>
            {DataPilem.shows.filter(show =>
                `${show.title} ${show.description}`.toUpperCase().indexOf(this.state.searchTerm.toUpperCase()) >= 0)
               .map(show => (
            <ShowCard key={show.imdbID} {...show} />
          ))}
        </div>
      </div>
    );
  }
}

export default Search;
