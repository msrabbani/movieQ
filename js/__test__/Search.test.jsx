// @flow 

import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { Unwrapped as UnwrappedSearch } from '../Search';
import showCard  from '../ShowCard';
import DataPilem from '../../data.json';

Enzyme.configure({ adapter: new Adapter() });
// console.log(process.env.NODE_ENV)

test('Search renders correctly', ()=> {
    const component = shallow(<UnwrappedSearch shows={DataPilem.shows} searchTerm='' />)
    expect(component).toMatchSnapshot()
})

test('Search should render correct amount of shows', () => {
    const component = shallow(<UnwrappedSearch shows={DataPilem.shows} searchTerm='' />)
    expect(component.find(showCard).length).toEqual(DataPilem.shows.length)
})

test('Search should render correct amount of shows based on search term', () => {
    const searchWord = 'black';
    const component = shallow(<UnwrappedSearch shows= {DataPilem.shows} searchTerm = {searchWord}/>)
    // component.find('input').simulate('change', {target: {value: searchWord}})

    const showCount = DataPilem.shows.filter(show => `${show.title} ${show.description}`.toUpperCase().indexOf(searchWord.toUpperCase()) >= 0).length

    expect(component.find(showCard).length).toEqual(showCount)
})
