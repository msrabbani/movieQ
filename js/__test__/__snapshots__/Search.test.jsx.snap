// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <header>
    <h1>
      video-Q
    </h1>
    <h1>
      
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Atlanta Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/a.jpg"
      />
      <div>
        <h1>
          Atlanta
        </h1>
        <h2>
          (
          2008–2013
          )
        </h2>
        <p>
          Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest 'Earn' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Billions Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/b.jpg"
      />
      <div>
        <h1>
          Billions
        </h1>
        <h2>
          (
          2016–
          )
        </h2>
        <p>
          U.S. Attorney Chuck Rhoades goes after hedge fund king, Bobby "Axe" Axelrod in a battle between two powerful New York figures.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Black Mirror Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/bm.jpg"
      />
      <div>
        <h1>
          Black Mirror
        </h1>
        <h2>
          (
          2011–
          )
        </h2>
        <p>
          A television anthology series that shows the dark side of life and technology.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Breaking Bad Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/bb.jpg"
      />
      <div>
        <h1>
          Breaking Bad
        </h1>
        <h2>
          (
          2008–2013
          )
        </h2>
        <p>
          A high school chemistry teacher diagnosed with inoperable lung cancer turns to manufacturing and selling methamphetamine in order to secure his family's future.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Game of Thrones Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/got.jpg"
      />
      <div>
        <h1>
          Game of Thrones
        </h1>
        <h2>
          (
          2011–
          )
        </h2>
        <p>
          Nine noble families fight for control over the mythical lands of Westeros, while a forgotten race returns after being dormant for thousands of years.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Homeland Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/h.jpg"
      />
      <div>
        <h1>
          Homeland
        </h1>
        <h2>
          (
          2011–
          )
        </h2>
        <p>
          A bipolar CIA operative becomes convinced a prisoner of war has been turned by al-Qaeda and is planning to carry out a terrorist attack on American soil.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="House of Cards Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/hoc.jpg"
      />
      <div>
        <h1>
          House of Cards
        </h1>
        <h2>
          (
          2013–
          )
        </h2>
        <p>
          A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Master of None Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/mon.jpg"
      />
      <div>
        <h1>
          Master of None
        </h1>
        <h2>
          (
          2015–
          )
        </h2>
        <p>
          The personal and professional life of Dev, a 30-year-old actor in New York.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Narcos Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/n.jpg"
      />
      <div>
        <h1>
          Narcos
        </h1>
        <h2>
          (
          2015–
          )
        </h2>
        <p>
          A chronicled look at the criminal exploits of Colombian drug lord Pablo Escobar.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Orange Is the New Black Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/oitnb.jpg"
      />
      <div>
        <h1>
          Orange Is the New Black
        </h1>
        <h2>
          (
          2013–
          )
        </h2>
        <p>
          The story of Piper Chapman, a woman in her thirties who is sentenced to fifteen months in prison after being convicted of a decade-old crime of transporting money for her drug-dealing girlfriend.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Silicon Valley Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/sv.jpg"
      />
      <div>
        <h1>
          Silicon Valley
        </h1>
        <h2>
          (
          2014–
          )
        </h2>
        <p>
          Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Stranger Things Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/st.jpg"
      />
      <div>
        <h1>
          Stranger Things
        </h1>
        <h2>
          (
          2016–
          )
        </h2>
        <p>
          When a young boy disappears, his mother, a police chief, and his friends must confront terrifying forces in order to get him back.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="The Americans Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/ta.jpg"
      />
      <div>
        <h1>
          The Americans
        </h1>
        <h2>
          (
          2013–
          )
        </h2>
        <p>
          Two Soviet intelligence agents pose as a married couple to spy on the American government.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Veep Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/v.jpg"
      />
      <div>
        <h1>
          Veep
        </h1>
        <h2>
          (
          2012–
          )
        </h2>
        <p>
          Former Senator Selina Meyer finds that being Vice President of the United States is nothing like she hoped and everything that everyone ever warned her about.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ivoqyK"
    >
      <img
        alt="Westworld Show Poster"
        className="sc-bwzfXH cOBXfF"
        src="public/img/posters/ww.jpg"
      />
      <div>
        <h1>
          Westworld
        </h1>
        <h2>
          (
          2016–
          )
        </h2>
        <p>
          Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence.
        </p>
      </div>
    </div>
  </div>
</div>
`;
